src/edge/simulator.ts

```diff
@@ -226,10 +191,168 @@ export class Simulator {
         });
     }
 
+    private async getLastestStandaloneSimulatorInfo() {
+        if (!this.latestStandaloneSimulatorInfoJson) {
+            const infoString = await request.get(Simulator.latestReleaseInfoUrl, {
+                headers: {
+                    "User-Agent": "vscode-azure-iot-edge",
+                },
+            });
+            const infoJson = JSON.parse(infoString);
+            this.latestStandaloneSimulatorInfoJson = infoJson;
+            return infoJson;
+        } else {
+            return this.latestStandaloneSimulatorInfoJson;
+        }
+    }
+
+    private async getSimulatorType(): Promise<SimulatorType> {
+        if (await this.simulatorInstalled()) {
+            if (Simulator.currentPlatform === "win32") {
+                return SimulatorType.Standalone;
+            } else {
+                return SimulatorType.Pip;
+            }
+        }
+        return SimulatorType.NotInstalled;
+    }
+
+    private async getLatestSimulatorVersion(): Promise<string | undefined> {
+        try {
+            let version: string;
+            if (Simulator.currentPlatform === "win32") {
+                const releaseInfoJson: any = await this.getLastestStandaloneSimulatorInfo();
+                version = releaseInfoJson.tag_name;
+            } else {
+                const pipResponse: string = await request.get(Simulator.iotedgehubdevVersionUrl);
+                version = JSON.parse(pipResponse).info.version;
+            }
+            return version;
+        } catch (error) {
+            return undefined;
+        }
+    }
+
+    private getUserConfiguredVersion(): string {
+        const version: string = Configuration.getConfigurationProperty("simulator.version");
+        if (version && version.match(/^\d+\.\d+\.\d+$/)) {
```

Comment:
only allowed number in version?

Reply:
for test package, we may have other chars

Reply:
oh, yes, I will use loose check to allow rc release